src/utils/__tests__/app-storage-test.js

```diff
@@ -0,0 +1,27 @@
+import { AppStorage } from "../app-storage";
+
+describe("AppStorage", () => {
+  const appStorage = new AppStorage();
+  test("AppStorage setItem", () => {
+    appStorage.setItem("test1", "storage1");
+    expect(appStorage.getItem("test1")).toBe("storage1");
+    appStorage.setItem("test2", JSON.stringify(["storage2, storage3"]));
+    var test2Result = JSON.parse(appStorage.getItem("test2"));
+    expect(test2Result).toStrictEqual(["storage2, storage3"]);
+    appStorage.removeItem("test1");
+    expect(appStorage.getItem("test1")).toBeNull();
+    appStorage.removeItem("test2");
+    expect(appStorage.getItem("test2")).toBeNull();
+
+    localStorage = null;
```

Comment:
let's put this in its own test() method, e.g. test("AppStorage with null localStorage", ...);